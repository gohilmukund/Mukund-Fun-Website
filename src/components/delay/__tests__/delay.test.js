import React from "react"
import { render, waitFor } from "@testing-library/react"
import Delay from "../index"

describe("Delay component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Delay wait={1} showLoadingAnimation={true}>
        Loaded
      </Delay>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it("execute the callback", async () => {
    const callback = jest.fn()
    const text = "Loaded"
    const { getByText } = render(
      <Delay wait={1} cb={callback} showLoadingAnimation={true}>
        <div>{text}</div>
      </Delay>
    )
    await waitFor(() => getByText(text))
    expect(callback).toHaveBeenCalled()
  })

  it("add a child element after a 1 second timeout", async () => {
    const text = "Loaded"
    const { getByText } = render(
      <Delay wait={1000} showLoadingAnimation={false}>
        <div>{text}</div>
      </Delay>
    )
    await waitFor(() => getByText(text))
    expect(getByText(text)).toHaveTextContent(text)
  })
})
