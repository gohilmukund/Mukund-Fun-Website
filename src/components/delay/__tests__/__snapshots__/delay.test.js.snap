// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delay component renders correctly 1`] = `
<div
  css="delayLoading"
/>
`;
